ivy/__init__.py

```diff
@@ -202,17 +202,37 @@ def __new__(cls, shape_tup):
                 current_backend(shape_tup).NativeArray,
             )
         ivy.utils.assertions.check_isinstance(shape_tup, valid_types)
-        if isinstance(shape_tup, int):
-            shape_tup = (shape_tup,)
+        if isinstance(shape_tup, ivy.NativeShape):
+            self._shape = shape_tup
+        elif isinstance(shape_tup, int):
+            self._shape = (shape_tup,)
         elif isinstance(shape_tup, list):
-            shape_tup = tuple(shape_tup)
-        ivy.utils.assertions.check_all(
-            [isinstance(v, int) or ivy.is_int_dtype(v.dtype) for v in shape_tup],
-            "shape must take integers only",
+            self._shape = tuple(shape_tup)
+        else:
+            self._shape = None
+
+    def __repr__(self):
+        return (
+            f"Ivy.Shape{self._shape}" if self._shape is not None else "Ivy.Shape(None)"
         )
-        if ivy.shape_array_mode():
-            return ivy.array(shape_tup)
-        return tuple.__new__(cls, shape_tup)
+
+    def __dir__(self):
+        self._shape.__dir__()
+
+    def __getattribute__(self, item):
+        return super().__getattribute__(item)
+
+    def __getitem__(self, key):
+        return self._shape[key] if self._shape is not None else None
+
+    def __setattr__(self, key, value):
+        if self._shape is not None:
+            self._shape[key] = value
```

Comment:
this would be the implementation of `setitem` and not `setattr` I guess. On top of that, I think this would fail with jax and numpy, as their shapes are tuples and therefore don't support item assignment directly